### PR TITLE
add proxy to versioned websites

### PR DESCRIPTION
**Pull request by ayj**
Proxy istio.io/v-${RELEASE} to istiodocs-v01.firebaseapp.com/v-${RELEASE}. 
This is generalized to support future release versions and assumes a 
consistent v-{major}.{minor} naming scheme.

Fixes frontend proxying for https://github.com/istio/istio.github.io/issues/379

### CODE REVIEW

File: istio.io/configmap-nginx.yaml [hunk 1]
```diff
@@ -84,6 +84,14 @@ data:
           return 200 "ok\n";
         }
 
+        # Proxy to past versioned instances of the site.
+        location ~ ^/v-[0-9]+\.[0-9]+.*$ {
+          proxy_set_header X-Real-IP $remote_addr;
+          proxy_set_header X-Forwarded-For $remote_addr;
+          proxy_cache      off;
+          proxy_pass       https://istiodocs-v01.firebaseapp.com:443;
```
Comment (PiotrSikora) [inline istio.io/configmap-nginx.yaml]: This should be handled via `upstream` block in order to enable keepalive connections (see example for `upstream github { ... }`).

File: istio.io/configmap-nginx.yaml [hunk 2]
```diff
@@ -84,6 +84,14 @@ data:
           return 200 "ok\n";
         }
 
+        # Proxy to past versioned instances of the site.
+        location ~ ^/v-[0-9]+\.[0-9]+.*$ {
+          proxy_set_header X-Real-IP $remote_addr;
+          proxy_set_header X-Forwarded-For $remote_addr;
+          proxy_cache      off;
```
Comment (PiotrSikora) [inline istio.io/configmap-nginx.yaml]: Presumably, versioned content doesn't change and we can enable cache here?

File: istio.io/configmap-nginx.yaml [hunk 3]
```diff
@@ -84,6 +84,14 @@ data:
           return 200 "ok\n";
         }
 
+        # Proxy to past versioned instances of the site.
+        location ~ ^/v-[0-9]+\.[0-9]+.*$ {
```
Comment (PiotrSikora) [inline istio.io/configmap-nginx.yaml]: Missing:
- `proxy_set_header Host istiodocs-v01.firebaseapp.com`,
- `proxy_set_header Connection ""`,
- `proxy_ssl_name istiodocs-v01.firebaseapp.com`.

Ideally, we should also add whatever SSL CA Firebase is using and verify the SSL certificate.

**Comment by ayj** [comment]
Thanks for the comments. Updated per your suggestions - PTAL. 

I've tested this in the canary namespace. I'll roll it out to istio.io tomorrow after the PR is approved.

File: istio.io/README.md [hunk 4]
```diff
@@ -145,10 +145,15 @@ configuration and certs.
...
 
 	curl -O https://www.digicert.com/CACerts/DigiCertHighAssuranceEVRootCA.crt
+        curl -O https://www.geotrust.com/resources/root_certificates/certificates/GeoTrust_Global_CA.pem
+
 	openssl x509 -inform DER -in DigiCertHighAssuranceEVRootCA.crt -outform PEM -out DigiCertHighAssuranceEVRootCA.pem
-	kubectl create secret generic cacerts --from-file=DigiCertHighAssuranceEVRootCA.pem --dry-run -o yaml > secret-cacerts.yaml
+
+	kubectl create secret generic cacerts \
+                --from-file=DigiCertHighAssuranceEVRootCA.pem \
+                --from-rile=GeoTrust_Global_CA.pem \
```
Comment (PiotrSikora) [inline istio.io/README.md]: s/rile/file/

Also, spaces vs tabs.

File: istio.io/README.md [hunk 5]
```diff
@@ -145,10 +145,15 @@ configuration and certs.
...
 - Automatic method: *TODO*
 
-Trusted CA certificate for backend (istio.github.io) verification is downloaded
-from [DigiCert](https://www.digicert.com/digicert-root-certificates.htm)
-and stored as a kubernetes secret.
+Download trusted CA certificates for backend (e.g. istio.github.io)
+verification and store as a kubernetes secret.
 
 	curl -O https://www.digicert.com/CACerts/DigiCertHighAssuranceEVRootCA.crt
+        curl -O https://www.geotrust.com/resources/root_certificates/certificates/GeoTrust_Global_CA.pem
```
Comment (PiotrSikora) [inline istio.io/README.md]: Nit: spaces vs tab.

**Pull request merged**